### DISCUSSION TITLE
Error: Builder returned invalid routes

### DISCUSSION THREAD

**Discussion by eddyw** [vercel/vercel#4799] [Help] [answered]
Hi,

We started to get this error today with new deployments of a NextJS app to Vercel:
```markdown
Error: Builder returned invalid routes: ["Route at index 18 has invalid `src` regular expression \"^/_next/data/****/(?<\\[\\.\\.\\.slug\\]>[^/]+?)\\.json$\"."]
```
We're using optional catch all routes and we have experimental `redirects` in `next.config.js`
> next@9.4.5-canary.12

The projects builds and this error appears after `Uploading build outputs...` is logged. The projects build fine locally without errors.

I'd appreciate if someone could help me understand where this comes from or where to look at. For instance, I have no clue what `Route at index 18` is. This doesn't seem a NextJS error since we use the exact version `next@9.4.5-canary.12` for some time now and it's been working fine with all previous deployments.

**Comment by paulogdm** [comment]
Hello,

This seems like a platform issue, indeed. Please reach out to https://vercel.com/support/request for further assistance.

**Comment by pcwa-ahendricks** [comment]
I've had the same issue with using `fallback: true` in getStaticPaths with the Nextjs framework. The only workaround I've found is to use `fallback: false` instead. Has anyone else figured out a different workaround that works with `fallback: true`?

**Comment by eddyw** [comment, marked as answer]
@paulogdm thanks, I did so.
@pcwa-ahendricks yeah, I figure that out too. I opened an issue in NextJS repo: https://github.com/vercel/next.js/issues/14923
I was told by support that I may need to wait for this PR https://github.com/vercel/vercel/pull/4803/ 😛

**Reply by styfle** [reply]
PR #4803 has been released so you should see new deployments working correctly now, thanks!

**Comment by pcwa-ahendricks** [comment]
Thanks for update @eddyw . That PR looks promising.
